plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -60,16 +63,92 @@ export interface RouterOptions {
   reader: UrlReader;
   database: PluginDatabaseManager;
   catalogClient: CatalogApi;
+
   actions?: TemplateAction<any>[];
   taskWorkers?: number;
   taskBroker?: TaskBroker;
   additionalTemplateFilters?: Record<string, TemplateFilter>;
+  identity?: IdentityApi;
 }
 
 function isSupportedTemplate(entity: TemplateEntityV1beta3) {
   return entity.apiVersion === 'scaffolder.backstage.io/v1beta3';
 }
 
+function buildDefaultIdentityClient({
+  logger,
+}: {
+  logger: Logger;
+}): IdentityApi {
+  return {
+    getIdentity: async ({ request }: IdentityApiGetIdentityRequest) => {
+      const header = request.headers.authorization;
+
+      if (!header) {
+        return undefined;
+      }
+
+      try {
+        const token = header.match(/^Bearer\s(\S+\.\S+\.\S+)$/i)?.[1];
+        if (!token) {
+          throw new TypeError('Expected Bearer with JWT');
+        }
+
+        const [_header, rawPayload, _signature] = token.split('.');
+        const payload: JsonValue = JSON.parse(
+          Buffer.from(rawPayload, 'base64').toString(),
+        );
+
+        if (
+          typeof payload !== 'object' ||
+          payload === null ||
+          Array.isArray(payload)
+        ) {
+          throw new TypeError('Malformed JWT payload');
+        }
+
+        const sub = payload.sub;
+        if (typeof sub !== 'string') {
+          throw new TypeError('Expected string sub claim');
+        }
+
+        // Check that it's a valid ref, otherwise this will throw.
+        parseEntityRef(sub);
+
+        return {
+          identity: {
+            userEntityRef: sub,
+            ownershipEntityRefs: [],
+            type: 'user',
+          },
+          token,
+        };
+      } catch (e) {
+        logger.error(`Invalid authorization header: ${stringifyError(e)}`);
+        return undefined;
+      }
+    },
+  };
+}
+
+const getIdentity = async ({
+  request,
+  identity,
+  logger,
+}: {
+  request: express.Request;
+  identity: IdentityApi;
+  logger: Logger;
+}) => {
+  let callerIdentity = undefined;
+  try {
+    callerIdentity = await identity.getIdentity({ request });
+  } catch (e: any) {
+    logger.debug(`identity could not be determined: ${e.message}`);
```

Comment:
I don't think this is right. We aren't meant to swallow all errors no matter what, right? I think we were meant to use getIdentity (no try/catch, just let it do its job) if it were given, and otherwise behave like we did before, which indeed was forgiving (but that behavior is already encapsulated inside the function above). So I think that getIdentity as a whole is no longer needed

Reply:
Sure. done. Would you expect it to return a 403/401. As I've implemented it raises an error that is not being handled and surfaces a 500 error to the uesr.. thats not nice. However, I am not sure I should add handling for every api route. It would seem cumbersome.. maybe there is another approach?

Reply:
See below!